contracts/voting/dao-voting-cw721-staked/src/contract.rs

```diff
@@ -343,22 +337,57 @@ pub fn execute_claim_nfts(
     deps: DepsMut,
     env: Env,
     info: MessageInfo,
+    token_ids: Option<Vec<String>>,
 ) -> Result<Response, ContractError> {
-    let nfts = NFT_CLAIMS.claim_nfts(deps.storage, &info.sender, &env.block)?;
-    if nfts.is_empty() {
-        return Err(ContractError::NothingToClaim {});
-    }
+    let token_ids = match token_ids {
+        // attempt to claim all legacy NFTs
+        None => {
+            // claim all legacy NFTs
+            let legacy_nfts =
+                LEGACY_NFT_CLAIMS.claim_nfts(deps.storage, &info.sender, &env.block)?;
+
+            if legacy_nfts.is_empty() {
+                return Err(ContractError::NothingToClaim {});
+            }
+
+            legacy_nfts
+        }
+        // attempt to claim non-legacy NFTs
+        Some(token_ids) => {
+            let token_ids = if token_ids.is_empty() {
```

Comment:
An enum would make more sense here tbh

```rust
pub enum ClaimType {
    All {},
    Select { token_ids: Vec<String> }
}
```

Reply:
agreed. i made this decision so that it would be somewhat backwards compatible, but that doesn't serve much purpose here in the message. this really should be:

```rust
pub enum ClaimType {
    Legacy {},
    All {},
    Select { token_ids: Vec<String> }
}
```

Reply:
Right but the actual message type could be Option<ClaimType> to keep that backward compatibility

Reply:
ohhh i see, nice. well now i'm convinced that we should not make this message backwards compatible, since the function of the empty argument is now changing. we should force the caller to know about the behavior, as the UI needs to detect the contract version to make sure it's claiming the right stuff.

Reply:
In this case, maybe we can attempt to claim from both legacy and current system? Then we can just have All and Select variants.
Behavior for All: 
If the error of both is EmptyClaims, then return empty claims error. If only one, then continue with claims for populated version. If other error, then error. If both not error, then both claimed.

Reply:
i believe legacy claiming still needs to be separate. in the case that the combination of all legacy and non-legacy claims is too large to load at once, then legacy claims will still be blocked.

even if the user selectively claims all non-legacy claims that are outstanding, there may still be non-legacy claims going through the unstaking period whose expirations have to be loaded and checked.